major rework of HA backup doc

MBishop17 opened the pull request.

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+This section describes how to:
 
->**Note:** Commands for one-time snapshots and recurring snapshots are available only in RKE v0.1.7 and later.
+- Create backups of your high-availability Rancher install.
```
Maybe add a link to the HA install.

MBishop17 replied:
Since there are two HA install procedures, I'm going to omit this suggestion.

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
````diff
@@ -4,217 +4,234 @@ weight: 370
-INFO[0006] [etcd] Successfully started [etcd-snapshot-once] container on host [z.z.z.z]
-INFO[0006] Finished saving snapshot [rke_etcd_snapshot_2018-05-17T23:32:08+02:00] on all etcd hosts
-```
+Backing up your high-availability Rancher cluster is process that invovles completing multiple tasks.
````
Spelling error: `involves`

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+	Before starting, make sure you have the files needed to create backups.
 
-To schedule a recurring automatic etcd snapshot save, enable the `etcd-snapshot` service. `etcd-snapshot` runs in a service container alongside the `etcd` container. `etcd-snapshot` automatically takes a snapshot of etcd and stores them to its local disk in `/opt/rke/etcd-snapshots`.
+2.  [Take Snapshots](#2-take-snapshots)
```
`Take Snapshots of the etcd Database`

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+3.  [Store Snapshot(s) Externally](#3-store-snapshots-externally)
 
-- `snapshot`: Enables/disables etcd snapshot recurring service in the RKE cluster.
+	After taking your snapshots, move them to a safe location that won't be affected if your cluster encounters issues.
```
instead of `move them`, let's use `export them`

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+- rancher-cluster.yml
 
-## High Availablity Restoration
+	You'll need the RKE config file you used for Rancher install, `rancher-cluster.yml`. You created this file during your chosen installation scenario:
```
`you used to install your high availability Rancher install` - We want to clarify that this is for HA versus single node that uses the docker container.

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
````diff
@@ -4,217 +4,234 @@ weight: 370
-    user: ubuntu
-    role: [etcd]
-```
+To take recurring snapshots, enable the `etcd-snapshot` service, which is a service that's included with RKE. This service runs in a service container alongside the `etcd` container. You can enable this service by adding some code to `rancher-cluster.yml`.
````
We should add in the info from here about the etcd service: https://github.com/rancher/rke#etcd-recurring-snapshots in this section. 

Describing what's going to happen beforehand, adding in details about the different fields

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+To take recurring snapshots, enable the `etcd-snapshot` service, which is a service that's included with RKE. This service runs in a service container alongside the `etcd` container. You can enable this service by adding some code to `rancher-cluster.yml`.
 
-After running `rke up` you should be able to have a two node cluster, the next step is to run few pods on node1:
+**To Enable Recuring Snapshots:**
```
Recurring

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
````diff
@@ -4,217 +4,234 @@ weight: 370
+	./rke_darwin-amd64 up --config rancher-cluster.yml
+	# Linux
+	./rke_linux-amd64 up --config rancher-cluster.yml
+	```
````
@galal-hussein Does this affect the HA setup by running up again? 

@MBishop17 we should probably note that if you run up again, you want to make sure that the Rancher setup hasn't changed, otherwise Rancher might change

galal-hussein replied:
@deniseschannon running rke up again shouldn't affect the HA setup

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+
+### 3. Backup Snapshots to a Safe Location
+
+After taking the etcd snapshots, save them to a safe location so that they're unaffected if your cluster experiences a disaster scenrio. This location should be persistent.
```
instead of save, use `export`

typoe: scenario

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+
+	If one of your `etcd` nodes goes down, create a new node, and then pull the most recent `etcd` snapshot to that node.
+
+2. [Restore ETCD Database](#2-restore-etcd-database)
```
Don't use ETCD, use etcd

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+<br/>
+### 1. Create New Node and Pull Snapshot
+
+If one of your etcd nodes go down, you need to replace it with a new node, and then pull the most recent working to that node.
```
` most recent working to that node.` - it's missing a word

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+
+1. Create a new node of your choice—baremetal, on-prem virtual machine, cloud-based virtual machine, and so on. Provision it according to our [requirements]({{< baseurl >}}/rancher/v2.x/en/installation/ha-server-install/# host-requirements).
+
+2.  Log in to your new node using your preferred shell, such as PuTTy or a remote Terminal connection.
```
i know this is from another section, but let's stick with suggesting open terminal as you've done that for the rest of these backup docs.

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
````diff
@@ -4,217 +4,234 @@ weight: 370
+docker ps | grep calico | cut -f 1 -d " " | xargs docker restart
 ```
+
+### 2. Restore ETCD Database
````
etcd not ETCD

deniseschannon commented on content/rancher/v2.x/en/installation/backups-and-restoration/ha-backup-and-restoration/_index.md:
```diff
@@ -4,217 +4,234 @@ weight: 370
+
+To restore the most recent `etcd` snapshot on your new node, run RKE's `etcd snapshot-restore` command. This command reverts to any snapshot stored in `/opt/rke/etcd-snapshots` that you explicitly define. When you run `etcd snapshot-restore`, RKE removes the old etcd container if it still exists. To restore operations, RKE creates a new etcd cluster using the snapshot you choose.
+
+>**Important:** When restoring the etcd database, you must restore each etcd to the _same_ snapshot, this means the exact same copy, so to restore you have to copy the snapshot from one of the nodes to the others before doing the `etcd snapshot-restore`
```
Needs a `.`

`from one of the the nodes to *all of the other nodes* before running `rke etcd snapshot-restore`.

MBishop17 commented:
@deniseschannon: replies to your general comments:

1. changed it to code blocks.
2. I adamantly disagree. A) I think most people won't put that together without frustration first. B) Just from a UX perspective, I think people will appreciate both commands there. If you really want me to change it, just ping me in slack :)
3. Yes, I am going to point to this file in both HA install docs. That's next on my list.
4. I agree with this comment. I updated all the commands with `rke` in front.
